performance-harness/proposals/performance_testing_harness_stage_3.md

```diff
@@ -0,0 +1,107 @@
+# Performance Testing Harness Stage 3 Planning
+
+Roadmap for the development of a **Performance Testing Harness** -- Stage 3.
+
+Requirements are being curated for the development of a performance testing and evaluation harness here: [Performance Testing Harness Requirements & Planning & Staging](https://github.com/eosnetworkfoundation/product/tree/main/performance-harness/proposals)
+
+
+## Stage 3
+
+###   Multi-producer & "relay" or "canary" node configuration support (Carried over from Stage 2)
+    -   Add configurable network topology based on public networks
+        -   multiple producer nodes, each accessed through a dedicated relay/canary node
+        -   all relay nodes connected in mesh
+    -   Include configurable "relay" nodes
+        -   Specific configuration per node type
+    -   Expand reporting to include multiple nodes
+
+### Work with ENF to identify additional metrics that would be useful to publish (Carried over from Stage 2)
+    -   Support additional metrics that would be useful to publish
+
+### HTTP harness testing

```

Comment:
Keeping with the normal modes of operation them

This would serve as the "API" node taking transactions largely from the HTTP interface BUT I think it is important that we measure the impact of ingesting blocks from the P2P protocol as well.  I don't think these blocks need to reflect the transactions that came in (maybe they do?) but they should be proportional to the load.  So if the HTTP enpoint is taking in 2k tps, then the blocks should have 2ktps on average.

care should be taken to make sure whatever is providing the blocks is not the bottleneck so we get "clean" numbers representing expected performance of this class of node.